tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -471,52 +656,189 @@ public static String getMethodSignature(MethodDeclaration methodDeclaration) {
      *     the signature {@code "... addAll(@NotNull final T[] elements)"}</li>
      *     <li>We lose parameter names, which are replaced by "arg0", "arg1", etc.</li>
      * </ul>
+     * We consider three types of ResolvedMethodDeclaration's: JavaParserMethodDeclaration,
+     * ReflectionMethodDeclaration, and JavassistMethodDeclaration.
      */
-    public static String getMethodSignature(MethodUsage methodUsage) {
-        String methodDeclaration = methodUsage.getDeclaration().toString();
-        Matcher matcher = METHOD_SIGNATURE.matcher(methodDeclaration);
+    public static String getMethodSignature(
+            MethodUsage methodUsage
+    ) {
+        ResolvedMethodDeclaration methodDeclaration = methodUsage.getDeclaration();
+        // Consider JavaParserMethodDeclaration.
+        if (methodDeclaration.toString().startsWith("JavaParserMethodDeclaration")) {
+            JavaParserMethodDeclaration jpMethodDeclaration = (JavaParserMethodDeclaration) methodDeclaration;
+            MethodDeclaration jpMethod = jpMethodDeclaration.getWrappedNode();
+            return getMethodSignature(jpMethod);
+        }
+        // Consider ReflectionMethodDeclaration or JavassistMethodDeclaration
+        Matcher matcher = METHOD_SIGNATURE.matcher(methodDeclaration.toString());
         if (!matcher.find()) {
             throw new IllegalStateException("Could not parse method signature: " + methodDeclaration);
         }
-        String methodModifiers = methodDeclaration.startsWith("ReflectionMethodDeclaration") ? matcher.group(1) : matcher.group(2);
-        // Take into account the case in which the method declaration refers to a method without an access specifier
+        String methodModifiers = methodDeclaration.toString().startsWith("ReflectionMethodDeclaration") ? matcher.group(1) : matcher.group(2);
```

Comment:
I would use `instanceof` rather than `toString()` and then a string test.  If that implementation approach is not possible, it would be good to document why.